rosette/lib/profile/record.rkt

```diff
@@ -113,25 +113,23 @@
 ;; The syntax rule generates the instrumentation for the two different
 ;; cases (keyword arguments or not), which helps expose some fusion
 ;; optimization opportunities to the compiler.
-(define-syntax-rule (runner app-proc-to-in-expr)
-  (let* ([handler  (lambda (exn) (values exn null))]
-         [returner (lambda e (values #f e))])
-    (lambda (loc proc in)
-      (record-enter! loc proc in)
-      (call-with-exception-handler
-       (lambda (e)
-         (record-exit! '())
-         e)
-       (thunk
-        (define ret (call-with-values (lambda () app-proc-to-in-expr) list))
-        (record-exit! ret)
-        (apply values ret))))))
+(define (runner proc*)
+  (lambda (loc proc in)
+    (record-enter! loc proc in)
+    (call-with-exception-handler
+     (lambda (e)
+       (record-exit! '())
+       e)
+     (thunk
+      (define ret (call-with-values proc* list))
+      (record-exit! ret)
+      (apply values ret)))))
```

Comment:
The macro here was intentional -- at the time we changed to this, the compiler wasn't great at optimizing keyword arguments through thunks, which made some benchmarks very slow. Can you test whether profiling performance regresses with this change? If not, fix the comment here which is now out of date.

Reply:
I will just revert it back.